source/views/sis/components/results.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import React from 'react'
+import {Text, View, StyleSheet, TouchableOpacity, Platform} from 'react-native'
+import {CourseType, parseTerm} from '../../../lib/course-search'
+import {Cell, TableView, Section} from 'react-native-tableview-simple'
+import {FormattedLine} from './formatted-line'
+import Icon from 'react-native-vector-icons/Ionicons'
+import * as c from '../../components/colors'
+
+type Props = {
+	terms: Array<{title: string, data: Array<CourseType>}>,
+}
+
+export default class CourseSearchTableView extends React.PureComponent<
+	Props,
+	State,
+> {
+	onPress = (course: CourseType) => {
+		this.props.navigation.navigate('CourseDetailView', {course: course})
+	}
+
+	renderCells = term => {
+		const cells = term.data.map(course => (
+			<TouchableOpacity
+				key={course.clbid}
+				onPress={() => {
+					this.onPress(course)
+				}}
+			>
+				<CourseCell course={course} />
+			</TouchableOpacity>
+		))
+		return cells
+	}
+
+	renderSection = term => {
+		let cells = this.renderCells(term)
+		let termHeader = parseTerm(term.title)
+		return (
+			<Section key={term.title} header={termHeader}>
+				{cells}
+			</Section>
+		)
+	}
+
+	render() {
+		// console.log(this.props.terms)
+		const sections = this.props.terms.map(term => this.renderSection(term))
+		return <TableView>{sections}</TableView>
+	}
+}
+
+const CourseCell = props => {
```

Comment:
You might want to look at the source/views/components/list folder; there's a Row that renders things in a row, a Column that does likewise, a Title that renders things like a platform-native table cell title, and also a Detail that does likewise for details. 

Those are what build up the Menu, Calendar, Dictionary, Orgs, Student Work, and even the Bus lists. I think they'd clean up a good amount of some of these visual components.

Reply:
Oh, and there's ListRow, and Separator, which handle spacing and visual separators on Android and iOS lists, as well as ListHeader(sp?) which does the same for section headers.